Firestore/core/src/firebase/firestore/util/async_queue.h

```diff
@@ -0,0 +1,204 @@
+   * queue, since this generally indicates a bug (and can lead to re-ordering of
+   * operations, etc).
+   *
+   * @param operation The operation to run.
+   */
+  void Enqueue(const Operation& operation);
+
+  /**
+   * Unlike `Enqueue`, this method does not require you to dispatch to a
+   * different queue than the current one (thus it is equivalent to a raw
+   * `dispatch_async()`).
+   *
+   * This is useful, e.g. for dispatching to the user's queue directly from user
+   * API call (in which case we don't know if we're already on the user's queue
+   * or not).
+   *
+   * @param operation The operation to run.
+   */
+  void EnqueueAllowingSameQueue(const Operation& operation);
+
+  /**
+   * Schedules an operation after the specified delay.
+   *
+   * Unlike `Enqueue`, this method does not require you to dispatch to a
+   * different queue than the current one.
+   *
+   * The returned `DelayedOperation` handle can be used to cancel the operation
+   * prior to its running.
+   *
+   * @param operation The operation to run.
+   * @param delay The delay after which to run the operation.
+   * @param timer_id A `TimerId` that can be used from tests to check for the
+   *     presence of this operation or to schedule it to run early.
+   * @return A `DelayedOperation` instance that can be used for cancellation.
+   */
+  DelayedOperation EnqueueAfterDelay(Milliseconds delay,
+                                     TimerId timer_id,
+                                     Operation operation);
+
+  /**
+   * Wrapper for `dispatch_sync()`. Mostly meant for use in tests.
+   *
+   * @param operation The operation to run.
+   */
+  void RunSync(const Operation& operation);
+
+  /**
+   * For tests: determine if a delayed operation with a particular `TimerId`
+   * exists.
+   */
+  bool ContainsDelayedOperation(TimerId timer_id) const;
+
+  /**
+   * For tests: runs delayed operations early, blocking until completion.
+   *
+   * @param last_timer_id Only delayed operations up to and including one that
+   *     was scheduled using this TimerId will be run. Method crashes if no
+   *     matching operation exists.
+   */
+  void RunDelayedOperationsUntil(TimerId last_timer_id);
+
+  /** The underlying wrapped `dispatch_queue_t`. */
+  dispatch_queue_t dispatch_queue() const {
+    return dispatch_queue_;
+  }
+
+ private:
+  void Dispatch(const Operation& operation);
+
+  void RemoveDelayedOperation(const detail::DelayedOperationImpl& operation);
+
+  bool OnTargetQueue() const;
+  // GetLabel functions are guaranteed to never return a "null" string_view
+  // (i.e. data() != nullptr).
+  absl::string_view GetCurrentQueueLabel() const;
+  absl::string_view GetTargetQueueLabel() const;
+
+  dispatch_queue_t dispatch_queue_{};
+  using OperationPtr = std::shared_ptr<detail::DelayedOperationImpl>;
+  std::vector<OperationPtr> operations_;
+  bool is_operation_in_progress_{};
```

Comment:
might just explicit set the initial value `true` or `false` than default initialize {}, which do have defined meaning of set to `false` but harder for human to read.